### DISCUSSION TITLE
Using your code in a non profit article

### DISCUSSION THREAD

**Discussion by timdf911** [Bodmer/TFT_eSPI#2323] [General]
Hi Bodmer
I tried to contact you off line but failed, so hope this is an appropriate question to pose here.
I've been working on a project for my own interest for over a year and making extensive use of your TFT libraries for graphics without which I would have been unable to proceed.
 One of my friends suggested I publish the design in a hobbyist magazine, which I have now done but have not released the code due to concerns about copyright etc.
This is not a commercial enterprise just one enthusiast trying to help others.
With the above in mind I'm seeking your permission to release the code and include credits and links within the code to your github.

Regards Tim

**Comment by Bodmer** [comment]
Yes, go ahead and publish! It's  great to hear about projects like this. You can also post project links here if you wish.

**Reply by timdf911** [reply]
Many thanks for such a quick positive reply.
I wish I could contribute more but my coding skills are somewhat limited.
Please accept my never ending thanks for providing such a useful resource to hobbyists such as myself.
Also please advise how I can 'buy you a coffee '.

Regards Tim
